backend/config/corpora-api.yml

```diff
@@ -492,11 +492,85 @@ paths:
         "404":
           $ref: "#/components/responses/404"
 
+  /v1/login:
+    get:
+      tags:
+        - authentication
+      summary: login to the system
+      operationId: corpora.lambdas.api.v1.authentication.login
+      responses:
+        "200":
+          $ref: "#/components/responses/200"
+        "401":
+          $ref: "#/components/responses/401"
+        "403":
+          $ref: "#/components/responses/403"
+
+  /v1/logout:
+    get:
+      tags:
+        - authentication
+      summary: logout of the system
+      operationId: corpora.lambdas.api.v1.authentication.logout
+      responses:
+        "200":
+          $ref: "#/components/responses/200"
+        "401":
+          $ref: "#/components/responses/401"
+        "403":
+          $ref: "#/components/responses/403"
+
+  /v1/userinfo:
+    get:
+      tags:
+        - authentication
+      summary: retrieve user information if the user is logged in
+      operationId: corpora.lambdas.api.v1.authentication.userinfo
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/userinfo"
+          $ref: "#/components/responses/200"
+        "401":
+          $ref: "#/components/responses/401"
+        "403":
+          $ref: "#/components/responses/403"
+
+  /v1/oauth2/callback:
+    get:
+      tags:
+        - authentication
+      summary: callback function from auth server after login
+      operationId: corpora.lambdas.api.v1.authentication.oauth2_callback
+      responses:
+        "200":
+          $ref: "#/components/responses/200"
+        "401":
+          $ref: "#/components/responses/401"
+        "403":
+          $ref: "#/components/responses/403"
+
 components:
   schemas:
     user_uuid:
       description: A unique identifier of a logged in User of Corpora.
       type: string
+    userinfo:
+      type: object
+      properties:
+        is_authentifacted:
+          type: bool
+        id:
+          $ref: "#/components/schemas/project_uuid"
```

Comment:
```suggestion
          $ref: "#/components/schemas/user_uuid"
```

Reply:
is this the user id from auth0 or from our database? We might want to use the same one for both, or add the auth0 id as a column in the user table.

Reply:
It's the id from auth0.  I assumed we would use that as the id for the the database too.   I assumed when we populate the database with users then we would use this as the key.